src/Swarm/Game/Scenario.hs

```diff
@@ -224,16 +257,28 @@ instance FromJSONE EntityMap Scenario where
     Scenario
       <$> liftE (v .: "name")
       <*> liftE (v .:? "description" .!= "")
-      <*> liftE ((fmap . fmap . map) reflow (v .:? "goal"))
       <*> liftE (v .:? "creative" .!= False)
       <*> liftE (v .:? "seed")
       <*> pure em
       <*> withE em (v ..:? "recipes" ..!= [])
       <*> withE em (localE (,rsMap) (v ..: "world"))
       <*> pure rs
-      <*> liftE (v .:? "win")
+      <*> liftE (parseObjectives v)
       <*> liftE (v .:? "solution")
       <*> liftE (v .:? "stepsPerTick")
+   where
+    -- Parse either a list of objectives, or a single objective via
+    -- old-style 'win'/'goal' fields.
+    parseObjectives :: Object -> Parser [Objective]
+    parseObjectives v =
+      fromMaybe []
+        <$> liftA2
+          (<|>)
+          (v .:? "objectives")
+          ( do
+              g <- v .:? "goal" .!= []
+              (fmap . fmap) ((: []) . Objective g) (v .:? "win")
+          )
```

Comment:
If I did not see the type, I could not tell what this function does.

Do we need to support the old-style global goal&win syntax at all, since we do not save much on typing?

For new scenario writers, it would be better to have the YAML syntax simple with a minimum of exclusive alternatives.

Reply:
No, we don't need to support the old-style syntax, it is just a bunch of work to update them all :sweat_smile: .  But I think you're right that it would be better to just support one simple syntax.  I will update them and simplify this code.

Reply:
15 tedious minutes and many emacs macros later, I have updated all the existing test scenarios. :sweat_smile: